File: gprolog-1.3.0/interpreter/call.pl

main :- 

 consult( testenv),

 checkComplainAndAdjustExpected( 0),

 asserta( a(1)),
 ( call( a( X)) -> 1=X, testOk ; testFail),
 ( call( a, X) -> 1=X, testOk ; testFail),
 retractall( a(1)),

 asserta( a(1, 2)),
 ( call( a( X, Y)) -> 1=X, 2=Y, testOk ; testFail),
 ( call( a( X), Y) -> 1=X, 2=Y, testOk ; testFail),
 ( call( a, X, Y) -> 1=X, 2=Y, testOk ; testFail),
 retractall( a(1, 2)),


 % see also : apply.pl, call.pl, topic-assert-clause-retract.pl

 checkComplainAndAdjustExpected( 5),

 result( S), 
 format( S, []),

 true.

File: R-3.1.1/interpreter/call.R

source( 'testEnv.R')

te = TestEnv()

te$checkComplainAndAdjustExpected( 0)

te$test( all( c( 1, 2, 3) == eval( call( 'c', 1, 2, 3))))

dcl= eval( call( 'list', c=1, a=2, b=3))

te$test( all( c( 1, 2, 3) == dcl))

te$test( all( c( "c", "a", "b") == names( dcl)))

te$checkComplainAndAdjustExpected( 3)

# see also: do.call.R, topic-substitute-eval.R, call.R
# see also: call.R, topic-call-eval.R
# see also: 

print( te$result())

